RFS/src/main/java/com/rfs/common/http/GzipRequestTransformer.java

```diff
@@ -0,0 +1,80 @@
+package com.rfs.common.http;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.Deflater;
+import java.util.zip.GZIPOutputStream;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+@AllArgsConstructor
+@Slf4j
+public class GzipRequestTransformer implements RequestTransformer {
+    private static final String CONTENT_ENCODING_HEADER_NAME = HttpHeaderNames.CONTENT_ENCODING.toString();
+    private static final String GZIP_CONTENT_ENCODING_HEADER_VALUE = "gzip";
+    private static final int READ_BUFFER_SIZE = 256 * 1024;  // Arbitrary, 256KB
+
+    // Local benchmarks show 15% throughput improvement with this setting
+    private static final int COMPRESSION_LEVEL = Deflater.BEST_SPEED;
+
+    @Override
+    public Mono<TransformedRequest> transform(
+        String method,
+        String path,
+        Map<String, List<String>> headers,
+        Mono<ByteBuffer> body
+    ) {
+        return body.map(this::gzipByteBufferSimple).singleOptional().flatMap(bodyOp -> {
+            Map<String, List<String>> newHeaders = new HashMap<>(headers);
+            if (bodyOp.isPresent()) {
+                newHeaders.put(CONTENT_ENCODING_HEADER_NAME, List.of(GZIP_CONTENT_ENCODING_HEADER_VALUE));
+            }
+            return Mono.just(new TransformedRequest(newHeaders, Mono.justOrEmpty(bodyOp)));
+        });
+    }
+
+    @SneakyThrows
+    private ByteBuffer gzipByteBufferSimple(final ByteBuffer inputBuffer) {
+        var readbuffer = inputBuffer.duplicate();
+        var baos = new ByteArrayOutputStream();
+        try (GZIPOutputStream gzipOutputStream = new FastGzipOutputStream(baos, READ_BUFFER_SIZE, false)) {
```

Comment:
two questions - why not just use the simple single-argument constructor?  Why do you have a subclass of FastGzipOutputStream (setting the level in this method)?

Reply:
The Deflator in DeflaterOutputStream (extended in GZIPOutputStream) is protected so i cannot set the level without  the subclass